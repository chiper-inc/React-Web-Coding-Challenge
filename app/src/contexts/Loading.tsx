import React, { createContext, useContext, useState } from 'react';

interface LoadingContextData {
  loading: boolean;
  showLoading: () => void;
  hideLoading: () => void;
}

export const LoadingContext = createContext<LoadingContextData>(
  {} as LoadingContextData,
);

export const LoadingProvider: React.FC = ({ children }) => {
  const [currentLoading, setCurrentLoading] = useState<boolean>(false);

  const showLoading = (): void => setCurrentLoading(true);

  const hideLoading = (): void => setCurrentLoading(false);

  return (
    <LoadingContext.Provider
      value={{
        loading: currentLoading,
        showLoading,
        hideLoading,
      }}
    >
      {children}
    </LoadingContext.Provider>
  );
};

const useLoading = () => useContext(LoadingContext);

export default useLoading;
