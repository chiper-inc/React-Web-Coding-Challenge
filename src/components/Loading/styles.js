import styled from "styled-components";

export const Wrapper = styled.span`
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
