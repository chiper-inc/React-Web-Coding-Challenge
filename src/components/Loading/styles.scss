.loading {
  @apply mt-8 flex-col flex justify-center items-center ;

  .spinner {
    @apply animate-spin
            rounded-full
            h-32
            w-32
            border-t-2 
            border-b-2 
          border-yellow-500;
  }
  .text {
    @apply text-4xl mt-4;
  }
}
