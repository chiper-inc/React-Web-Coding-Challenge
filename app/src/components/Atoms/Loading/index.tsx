import React from 'react';
import { Spinner } from './styles';

const Loading: React.FC = () => <Spinner />;

export { Loading };
