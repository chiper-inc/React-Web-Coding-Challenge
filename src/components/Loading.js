import React from "react";
import { Typography, CircularProgress } from "@material-ui/core";
import { LoadingContainer } from "../styles/LoadingStyles";

export const Loading = () => {
  return (
    <LoadingContainer>
      <CircularProgress />
      <Typography variant="h3" component="h3" color="primary">
        Loading...
      </Typography>
    </LoadingContainer>
  );
};
