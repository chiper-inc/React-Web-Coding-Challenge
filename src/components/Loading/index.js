import React from 'react'
import { OrbitalSpinnerStyled } from './styles'

export const OrbitalSpinner = () => <OrbitalSpinnerStyled color='#343a40' />
