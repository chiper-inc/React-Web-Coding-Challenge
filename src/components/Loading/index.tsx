import { Wrapper } from "./styles";
import Puff from "../../assets/images/puff.svg";

export const Loading = () => {
  return (
    <Wrapper>
      <img src={Puff} alt="loading"/>
    </Wrapper>
  );
};
