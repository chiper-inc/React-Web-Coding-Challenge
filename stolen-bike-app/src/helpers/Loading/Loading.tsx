import { Spinner, WrapperState } from "./Loading.styles"

export const Loading = () => {
  return (
    <WrapperState>
      <Spinner className="fa-solid fa-circle-notch"></Spinner>
    </WrapperState>
  )
}
