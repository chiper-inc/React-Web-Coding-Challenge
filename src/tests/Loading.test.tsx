import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../components/Loading';
test('renders the component Loading', () => {
  const component = shallow(<Loading />);
  expect(component).toMatchSnapshot();
});
