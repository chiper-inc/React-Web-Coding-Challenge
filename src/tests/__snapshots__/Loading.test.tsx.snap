// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component Loading 1`] = `
<div
  className="loading"
>
  <div
    className="spinner"
  />
  <p
    className="text"
  >
    Loading
  </p>
</div>
`;
